x-pack/plugins/maps/public/layers/sources/source.ts

```diff
@@ -0,0 +1,195 @@
+  isTimeAware(): Promise<boolean>;
+  getImmutableProperties(): Promise<ImmutableSourceProperty[]>;
+  getAttributions(): Promise<Attribution[]>;
+  isESSource(): boolean;
+  renderSourceSettingsEditor({ onChange }: { onChange: () => void }): ReactElement<any> | null;
+  supportsFitToBounds(): Promise<boolean>;
+  isJoinable(): boolean;
+  cloneDescriptor(): SourceDescriptor;
+  getFieldNames(): string[];
+  getApplyGlobalQuery(): boolean;
+  getIndexPatternIds(): string[];
+  getQueryableIndexPatternIds(): string[];
+  getGeoGridPrecision(zoom: number): number;
+  shouldBeIndexed(): boolean;
+  getPreIndexedShape(): Promise<PreIndexedShape | null>;
+  createFieldFormatter(field: IField): Promise<FieldFormatter | null>;
+  loadStylePropsMeta(args: unknown): Promise<unknown>; // todo
+  getValueSuggestions(field: IField, query: string): Promise<string[]>;
+}
+
+export class AbstractSource implements ISource {
+  static isIndexingSource = false;
+
+  readonly _descriptor: SourceDescriptor;
+  readonly _inspectorAdapters?: object;
+
+  constructor(descriptor: SourceDescriptor, inspectorAdapters?: object) {
+    this._descriptor = descriptor;
+    this._inspectorAdapters = inspectorAdapters;
+  }
+
+  destroy(): void {}
+
+  cloneDescriptor(): SourceDescriptor {
+    // @ts-ignore
+    return copyPersistentState(this._descriptor);
+  }
+
+  async supportsFitToBounds(): Promise<boolean> {
+    return true;
+  }
+
+  /**
+   * return list of immutable source properties.
+   * Immutable source properties are properties that can not be edited by the user.
+   */
+  async getImmutableProperties(): Promise<ImmutableSourceProperty[]> {
+    return [];
+  }
+
+  getInspectorAdapters(): object | undefined {
+    return this._inspectorAdapters;
+  }
+
+  createDefaultLayer(): ILayer {
+    throw new Error(`Source#createDefaultLayer not implemented`);
+  }
+
+  async getDisplayName(): Promise<string> {
+    return '';
+  }
+
+  /**
+   * return attribution for this layer as array of objects with url and label property.
+   * e.g. [{ url: 'example.com', label: 'foobar' }]
+   * @return {Promise<null>}
+   */
+  async getAttributions(): Promise<Attribution[]> {
+    return [];
+  }
+
+  isFieldAware(): boolean {
+    return false;
+  }
+
+  isRefreshTimerAware(): boolean {
+    return false;
+  }
+
+  isGeoGridPrecisionAware(): boolean {
+    return false;
+  }
+
+  isQueryAware(): boolean {
+    return false;
+  }
+
+  getFieldNames(): string[] {
+    return [];
+  }
+
+  renderSourceSettingsEditor() {
+    return null;
+  }
+
+  getApplyGlobalQuery(): boolean {
+    return !!this._descriptor.applyGlobalQuery;
+  }
+
+  getIndexPatternIds(): string[] {
+    return [];
+  }
+
+  getQueryableIndexPatternIds(): string[] {
+    return [];
+  }
+
+  getGeoGridPrecision(zoom: number): number {
+    return 0;
+  }
+
+  isJoinable(): boolean {
+    return false;
+  }
+
+  shouldBeIndexed(): boolean {
+    return AbstractSource.isIndexingSource;
```

Comment:
Just return false and remove static `AbstractSource.isIndexingSource`

Reply:
thanks. removing this from `GeoJsonFileSource` as well